### ISSUE TITLE
Trigger from declarative pipeline?

### ISSUE THREAD

**Issue by chrylis** [jenkinsci/pipeline-github-plugin#10]
As mentioned [in this comment](https://github.com/jenkinsci/pipeline-github-plugin/issues/6#issuecomment-368604132), is there a way to register the trigger when using a declarative pipeline?

**Comment by aaronjwhiteside** [comment]
Hi @chrylis I don't personally use the declarative pipeline syntax, but according to the documentation this should work.

```
pipeline {
    triggers {
        issueCommentTrigger('.*test this please.*')
    }
}
```

Let me know how this works for you..

**Comment by stefan-dimitrov** [comment]
@aaronjwhiteside, It doesn't seem to work for me. I have the Jenkinsfile like that:
```
pipeline {

  agent {
    node {
      label 'docker'
    }
  }

  stages {
    ...
  }

  triggers {
    issueCommentTrigger('.*retest this please.*')
  }
}
```

I've made a PR adding the `triggers` section. First build ran ok.
When I add a comment `test this please` or `retest this please` in the same PR, nothing happens - no build is triggered.

**Comment by stefan-dimitrov** [comment]
I see 2 commented lines in `GitHubEventSubscriber.java`:

![screenshot from 2018-04-11 15-03-57](https://user-images.githubusercontent.com/6047637/38615517-9fb00fc6-3d99-11e8-8cfa-8d751fafd2ec.png)

Maybe these should be uncommented ?

**Comment by aaronjwhiteside** [comment]
So you're correct, it currently only support `ISSUE_COMMENT`'s when I first implemented this, the limitation was due to the fact that the GitHub Plugin ignored the other events, so even if we signaled were were interested in receiving them, we still wouldn't get them.

I'm currently trying to find the class within the GitHub Plugin where this was done, maybe/hopefully this restriction has been lifted now? When I find the class/place I'll comment here again.

But thought I should give you some response, until then..

**Comment by stefan-dimitrov** [comment]
We got it to work (without changes to jenkinsfile or the plugin) - we added the web hooks for `Issue comments` and `Pull request review comments` to our GH repo.

<img width="695" alt="gh-repo-webhoooks" src="https://user-images.githubusercontent.com/6047637/38663916-dd4dfe86-3e40-11e8-9ceb-7069a28e478c.png">

**Comment by hosseinkol** [comment]
@stefan-dimitrov Are you using a declarative pipeline? Could you post a snippet of your Jenkinsfile? I've setup the web-hook per your suggestion but cannot triggers builds. Here is the begining of my pipeline block:

`pipeline {
    agent {
      label {
        label "master"
        customWorkspace "${JENKINS_HOME}/workspace/myrepo/${BRANCH_NAME}"
      }
    }
    triggers {
     issueCommentTrigger('.*/test.*')
    }`

**Comment by stefan-dimitrov** [comment]
@hosseinkol, here is a snippet from my jenkinsfile:
```
pipeline {
  triggers {
    issueCommentTrigger('.*retest this please.*')
  }

  agent {
    node {
      label 'docker'
    }
  }

  stages {
    // Rest of the Jenkinsfile
```

**Comment by hosseinkol** [comment]
@stefan-dimitrov I've noticed that despite github sending issue_comment events to Jenkins & receiving a 200 response, Jenkins logs show no recorded event of type ISSUE_COMMENT. However events of type PULL_REQUEST are fine. I'm not sure whats blocking Jenkins from receiving the ISSUE_COMMENT.

**Comment by eplodn** [comment]
@stefan-dimitrov Doesn't work here either.

I've got a GitHub webhook implemented as above, I can see the deliveries.

My Jenkinsfile:

```
pipeline {
    triggers {
       issueCommentTrigger('.*pipeline please rebuild.*')
     }
```

I have a log defined for `org.jenkinsci.plugins.pipeline.github.trigger.IssueCommentTrigger`, it comes up empty.

My installation is also using GHPRB plugin, it catches the issue comments; could it be that GHPRB is preventing pipeline-github from receiving them?

```Pull request #5,112 was updated/initialized on drivenets/cheetah at 7/10/18 10:44 AM by eplodn  (PR update)
Jul 10, 2018 10:45:08 AM FINEST org.jenkinsci.plugins.ghprb.GhprbPullRequest
Checking for comments after: 7/10/18 10:04 AM
Jul 10, 2018 10:45:08 AM FINEST org.jenkinsci.plugins.ghprb.GhprbPullRequest
Comment was made at: 7/10/18 10:44 AM
Jul 10, 2018 10:45:08 AM FINEST org.jenkinsci.plugins.ghprb.GhprbPullRequest
Comment was made after last update time, please rebuild
Jul 10, 2018 10:45:08 AM FINEST org.jenkinsci.plugins.ghprb.GhprbPullRequest
[eplodn ] Added comment: pipeline please rebuild```

**Comment by cryptk** [comment]
Any movement on this? I would love to use this trigger functionality, but as far as I can tell from this issue, it doesn't work

**Comment by eplodn** [comment]
@cryptk FYI, I got it working eventually.

**Comment by tjhei** [comment]
> @cryptk FYI, I got it working eventually.

How did you get it to work? Would you mind sharing?

**Comment by eplodn** [comment]
@tjhei I think that once I've got my project defined as "GitHub organization" and made sure the webhooks worked (with reverse proxy), it started working.

**Comment by mleenhardt** [comment]
Anybody found a way to make this work? Using Jenkins 2.150.1 and the latest plugin (2.1), I'm not able to trigger builds by posting comments. This is my jenkinsfile

```
pipeline {
  triggers {
    issueCommentTrigger('.*retest.*')
  }
  agent any  
  stages {
        stage('Build') {
            steps {
                sh 'echo "Hello world"'
            }
        }
    }
}
```

**Comment by njt1982** [comment]
What do we need to change to make that webhook enabled by default (eg if I onboard a new repository)?

**Comment by Jiaweizdev** [comment]
I always get `Invalid trigger type "issueCommentTrigger". Valid trigger types: [upstream, cron, githubPush, pollSCM]` when I try to use 
```
 triggers {
      issueCommentTrigger('.*test this please.*')
}
```
in declarative pipeline.

**Comment by rpassis** [comment]
I have the same issue as @Jiaweizdev - `Invalid trigger type "issueCommentTrigger". Valid trigger types: [upstream, cron, githubPush, pollSCM]`

Is there any configuration required to enable this on the Jenkins side?

**Comment by rpassis** [comment]
I had the same issue and it turns out the plugin needed to be installed 😬

**Comment by shivamsood** [comment]
which plugin @rpassis

**Comment by rpassis** [comment]
The `pipeline-github-plugin` i.e. the plugin that is available in this repo.

**Comment by adrienzieba** [comment]
FYI, after installing the plugin, if your github webhook are managed by jenkins, you need to Re-register them (`Jenkins->Manage->Github->Re-register hooks for all jobs`). It will update your github webhooks to add `Issue comments` event to the list of supported events.

**Comment by victtsl** [comment]
It worked on my Jenkins installation is using both GHPRB plugin and pipeline-github-plugin. I created a Github Org project and using sample Jenkinsfile from stefan-dimitrov and then I created a logger for GitHub in https://<JENKINS_URL>/log/ including:

- hudson.plugins.git.GitStatus ALL
- com.cloudbees.jenkins.GitHubWebHook - ALL
- org.jenkinsci.plugins.github.- ALL

Tested it by adding comments to PR and verified that similar outputs below are in the logs:
`Mar 04, 2021 5:04:21 PM FINE org.jenkinsci.plugins.github.webhook.GHEventHeader$PayloadHandler parse
Header X-GitHub-Event -> issue_comment`

`Mar 04, 2021 5:04:21 PM FINE org.jenkinsci.plugins.pipeline.github.trigger.GitHubEventSubscriber onEvent
Received event: ISSUE_COMMENT`

However GitHub Org project is still not triggered from comments. In desperation, I rebooted Jenkins server and it worked after that.

I used these links to help troubleshoot the issue as well:
https://support.cloudbees.com/hc/en-us/articles/224543927-GitHub-webhook-configuration
https://support.cloudbees.com/hc/en-us/articles/115003015711-GitHub-Webhook-Organization-Folder
https://support.cloudbees.com/hc/en-us/articles/224621648-GitHub-webhook-troubleshooting
